tests/PageDiaryEntryTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+require_once __DIR__ . '/_CustomTestRunnerHooks.php';
+use PHPUnit\Framework\TestCase;
+use WTSA1\Engines\Database;
+use WTSA1\Engines\Session;
+use WTSA1\Models\Category;
+use WTSA1\Models\User;
+use WTSA1\Pages\DiaryEntryPage;
+
+class PageDiaryEntryTestee extends DiaryEntryPage {
+    public function testPost() {
+        $this->post();
+    }
+
+    public function dumpData() {
+        return $this->data;
+    }
+}
+
+class PageDiaryEntryTest extends TestCase {
+
+    public function setUp(): void {
+        Session::getInstance()->setUser(User::register("test", "test"));
+        echo "Seeding category table...\r\n";
+        $dummyData = array(
+            array(1, "Nicht kategorisiert"),
+            array(2, "Ferien"),
+            array(3, "Geburtstag"),
+            array(4, "Familienfest"),
+            array(5, "Ausflug")
+        );
+        Database::getInstance()->query("
+            INSERT INTO `category` (id, category) VALUES
+            ". implode(", ", array_map(function($item) { return "('".$item[0]."','".$item[1]."')"; }, $dummyData)) . "
+            ;
+        ");
+    }
```

Comment:
Minor issue, wouldn't it be easier to only populate `(category)` and let the DB generate the ID? Then we can just write out the whole query as `INSERT INTO category (category) VALUES (Unkategorisiert), (Ferien)...;`

Reply:
I did it this way in case we have data that gets seed by the id of the category. Since PHP sorts the array as it wishes, i guess it was more reliable to do it this way.

Reply:
Ah good point, since we can have dependent seed data :+1: